packages/web-components/src/horizontalbarchart/horizontalbarchart.ts

```diff
@@ -0,0 +1,384 @@
+    function createBars(this: SVGGElement, point: IChartDataPoint, index: number) {
+      const _isRTL = false;
+      const barHeight = 12;
+      const pointData = point.data ? point.data : 0;
+      if (index > 0) {
+        prevPosition += value;
+      }
+      value = (pointData / total) * 100 ? (pointData / total) * 100 : 0;
+      if (value < 1 && value !== 0) {
+        value = 1 / scalingRatio;
+      } else if (value > 99 && value !== 100) {
+        value = 99 / scalingRatio;
+      } else {
+        value = value / scalingRatio;
+      }
+
+      startingPoint.push(prevPosition);
+
+      const gEle = d3Select(this) // 'this' refers to the current 'g' element
+        .attr('key', index)
+        .attr('role', 'img')
+        .attr('aria-label', pointData);
+
+      gEle
+        .append('rect')
+        .attr('key', index)
+        .attr('id', `${barNo}-${index}`)
+        .attr('barinfo', `${point.legend}`)
+        .attr('class', 'bar')
+        .attr('style', `fill:${point.color!}`)
+        .attr(
+          'x',
+          `${
+            _isRTL
+              ? 100 - startingPoint[index] - value - barSpacingInPercent * index
+              : startingPoint[index] + barSpacingInPercent * index
+          }%`,
+        )
+        .attr('y', 0)
+        .attr('width', value + '%')
+        .attr('height', barHeight)
+        .attr('tabindex', 0)
+        .attr('data-tabster', '{"groupper": {...}}"')
+        .attr('data-tabster', '{"mover": {...}}"');
+    }
+
+    const containerDiv = d3Create('div').attr('style', 'position: relative');
+
+    let tooltip: any;
+
+    const chartTitleDiv = containerDiv.append('div').attr('class', 'chartTitleDiv');
+    chartTitleDiv
+      .append('div')
+      .append('span')
+      .attr('class', 'chartTitle')
+      .text(data?.chartTitle ? data?.chartTitle : '');
+
+    const hideNumber = this.hideRatio === undefined ? false : this.hideRatio;
+
+    const showRatio = this.variant === Variant.PartToWhole && !hideNumber && data!.chartData!.length === 2;
+    const getChartData = () => (data!.chartData![0].data ? data!.chartData![0].data : 0);
+
+    if (showRatio) {
+      const ratioDiv = chartTitleDiv.append('div').attr('role', 'text');
+      const numData = data!.chartData![0].data;
+      const denomData = data!.chartData![1].data;
+      const total = numData! + denomData!;
+      ratioDiv.append('span').attr('class', 'ratioNumerator').text(numData!);
+      ratioDiv.append('span').attr('class', 'ratioDenominator').text(`/${total!}`);
+    }
+
+    const svgEle = containerDiv
+      .append('svg')
+      .attr('height', 20)
+      .attr('width', 100 + '%')
+      .attr('aria-label', data?.chartTitle ? data?.chartTitle : '')
+      .selectAll('g')
+      .data(data.chartData!)
+      .enter()
+      .append('g')
+      .each(createBars)
+      .on('mouseover', function (event, d) {
+        const tooltipHTML = `
+        <div style="border-left:4px solid ${d.color}; padding-left: 8px;">
```

Comment:
can we move these styles to the stylesheet?

Reply:
Also use css logical properties so instead of `padding-left` use `padding-inline-start`

Reply:
Tooltip html is moving to the template.